furusiya/model/components/walkers/random_walker.py

```diff
@@ -8,23 +8,16 @@ def __init__(self, area_map, parent):
         self.parent = parent
 
     def walk(self):
-        valid_options = [0, 1, 2, 3]
-        random.shuffle(valid_options)
+        # Not ideal, but better than what was here previously.
+        adjacent_tiles = [
+            (self.parent.x + x_offset, self.parent.y + y_offset)
+            for x_offset, y_offset in [(1, 0), (-1, 0), (0, 1), (0, -1)]
+        ]
+        random.shuffle(adjacent_tiles)
 
-        while valid_options:
-            x, y = self.parent.x, self.parent.y
-            n = valid_options.pop()
-
-            if n == 0:
-                x += 1
-            elif n == 1:
-                x -= 1
-            elif n == 2:
-                y += 1
-            elif n == 3:
-                y -= 1
-            if self.area_map.is_on_map(x, y) and self.area_map.is_walkable(x, y):
-                self.parent.x, self.parent.y = x, y
+        for x_, y_ in adjacent_tiles:
```

Comment:
you can call these `tile_x` or `adjacent_x` or something, `x_` looks weird to me.